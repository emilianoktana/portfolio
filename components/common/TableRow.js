import React from 'react'
import {
    View,
    Text,
    StyleSheet
} from 'react-native'

import { SELECTED_ROW_COLOR, WHITE_COLOR } from '../../constants/Colors'

export const TableRow = (props) => {

    const selectedBackground = (props.selected) ? SELECTED_ROW_COLOR : WHITE_COLOR
    
    return (
        <View style={{ flex: 1, alignSelf: 'stretch', flexDirection: 'row', backgroundColor: selectedBackground }}>
            {props.data.map((val, index) => 
                <View key={index} style={styles.cell}>
                    <Text>{ props.data[index] }</Text>
                </View>)
            }
        </View>
    )
}

const styles = StyleSheet.create({
    cell: {
        flex: 1, 
        alignSelf: 'stretch', 
        borderWidth: 1,
        padding: 5,
        alignItems: 'center'
    }
})