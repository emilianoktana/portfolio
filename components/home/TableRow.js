import React from 'react'
import {
    View,
    Text
} from 'react-native'

export const TableRow = (props) => {
    const selectedBackground = (props.selected) ? '#86ccf7' : "#fff"
    return (
        <View style={{ flex: 1, alignSelf: 'stretch', flexDirection: 'row', backgroundColor: selectedBackground }}>
            <View style={{ flex: 1, alignSelf: 'stretch', borderWidth: 1, borderRightWidth: 0 }}>
                <Text>{ props.data[0] }</Text>
            </View> 
            <View style={{ flex: 1, alignSelf: 'stretch', borderWidth: 1, borderRightWidth: 0 }}>
                <Text>{ props.data[1] }</Text>
            </View>
            <View style={{ flex: 1, alignSelf: 'stretch', borderWidth: 1, borderRightWidth: 0 }}>
                <Text>{ props.data[2] }</Text>
            </View>
            <View style={{ flex: 1, alignSelf: 'stretch', borderWidth: 1, borderRightWidth: 0 }}>
                <Text>{ props.data[3] }</Text>
            </View>
            <View style={{ flex: 1, alignSelf: 'stretch', borderWidth: 1 }}>
                <Text>{ props.data[4] }</Text>
            </View>
        </View>
    )
}