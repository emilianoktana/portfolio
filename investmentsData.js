export default {
    'investments': [
        'Bonds', 'Insurance', 'Retirement', 'Commodity', 'Annuities'
    ],
    'risks': {
        "1": [80, 20, 0, 0, 0],
        "2": [70, 15, 15, 0, 0],
        "3": [60, 15, 15, 10, 0],
        "4": [50, 20, 20, 10, 0],
        "5": [40, 20, 20, 20, 0],
        "6": [35, 25, 5, 30, 5],
        "7": [20, 25, 25, 25, 5],
        "8": [10, 20, 40, 20, 10],
        "9": [5, 15, 40, 25, 15],
        "10": [0, 5, 25, 30, 40]
    }
}