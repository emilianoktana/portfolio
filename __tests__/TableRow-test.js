/**
 * @jest-environment jsdom
 */
import React from 'react'
import { TableRow } from '../components/common/TableRow'

import investmentsData from '../investmentsData'
const items = investmentsData.investments

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<TableRow selected={false} data={items} />).toJSON()
  expect(tree).toMatchSnapshot()
});
