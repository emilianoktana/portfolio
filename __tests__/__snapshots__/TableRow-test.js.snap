// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "flex": 1,
        "flexDirection": "row",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderWidth": 1,
        "flex": 1,
        "padding": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Bonds
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderWidth": 1,
        "flex": 1,
        "padding": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Insurance
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderWidth": 1,
        "flex": 1,
        "padding": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Retirement
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderWidth": 1,
        "flex": 1,
        "padding": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Commodity
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderWidth": 1,
        "flex": 1,
        "padding": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Annuities
    </Text>
  </View>
</View>
`;
